exercises/practice/saddle-points/runtests.jl

```diff
@@ -0,0 +1,48 @@
+using Test
+
+include("saddle-points.jl")
+
+@testset "Can identify single saddle point" begin
+    matrix = [9 8 7; 5 3 2; 6 6 7]
+    @test saddlepoints(matrix) == [(2, 1)]
+end
+
+@testset "Can identify that empty matrix has no saddle points" begin
+    matrix = []
+    @test saddlepoints(matrix) == []
+end
+
+@testset "Can identify lack of saddle points when there are none" begin
+    matrix = [1 2 3; 3 1 2; 2 3 1]
+    @test saddlepoints(matrix) == []
+end
+
+@testset "Can identify multiple saddle points in a column" begin
+    matrix = [4 5 4; 3 5 5; 1 5 4]
+    @test saddlepoints(matrix) == [(1, 2), (2, 2), (3, 2)]
+end
+
+@testset "Can identify multiple saddle points in a row" begin
+    matrix = [6 7 8; 5 5 5; 7 5 6]
+    @test saddlepoints(matrix) == [(2, 1), (2, 2), (2, 3)]
+end
+
+@testset "Can identify saddle point in bottom right corner" begin
+    matrix = [8 7 9; 6 7 6; 3 2 5]
+    @test saddlepoints(matrix) == [(3, 3)]
+end
+
+@testset "Can identify saddle points in a non square matrix" begin
+    matrix = [3 1 3; 3 2 4]
+    @test saddlepoints(matrix) == [(1, 1), (1, 3)]
+end
+
+@testset "Can identify that saddle points in a single column matrix are those with the minimum value" begin
+    matrix = reshape([2; 1; 4; 1], :, 1)
```

Comment:
You can also write this as `[2; 1; 4; 1;;]` if you prefer. For some reason that's slower but it could be easier to figure out because one doesn't have to learn about `reshape`?

Reply:
```suggestion
    matrix = reshape([2, 1, 4, 1], :, 1)
```

There's no need for the semicolons. Vectors in Julia are already column vectors (and reshape replaces the old shape of the matrix anyway).

I think reshape is the standard way to do this. The technique with extra semicolons gives different answers for different Julia versions.

Reply:
> Vectors in Julia are already column vectors 

The example doesn't pass the tests when doing this, it needs the one column matrix for the indices to work. But that probably means the example should be changed, not the tests.

Reply:
I think it's fine for the example to require the input to be an AbstractMatrix and would be confusing for students if they had to support being given "matrices" that actually only have one dimension.

Reply:
```suggestion
    matrix = reshape([2, 1, 4, 1], :, 1)
```